src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/TaskPlanServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package ch.cern.eam.wshub.core.services.workorders.impl;
+
+import ch.cern.eam.wshub.core.client.InforContext;
+import ch.cern.eam.wshub.core.services.workorders.TaskPlanService;
+import ch.cern.eam.wshub.core.services.workorders.entities.TaskPlan;
+import ch.cern.eam.wshub.core.tools.ApplicationData;
+import ch.cern.eam.wshub.core.tools.InforException;
+import ch.cern.eam.wshub.core.tools.Tools;
+import net.datastream.schemas.mp_functions.mp0079_001.MP0079_GetTask_001;
+import net.datastream.schemas.mp_results.mp0079_001.MP0079_GetTask_001_Result;
+import net.datastream.wsdls.inforws.InforWebServicesPT;
+
+import java.math.BigInteger;
+
+public class TaskPlanServiceImpl implements TaskPlanService {
+    private Tools tools;
+    private InforWebServicesPT inforws;
+    private ApplicationData applicationData;
+
+    public TaskPlanServiceImpl(ApplicationData applicationData, Tools tools, InforWebServicesPT inforWebServicesToolkitClient){
+        this.applicationData = applicationData;
+        this.tools = tools;
+        this.inforws = inforWebServicesToolkitClient;
+    }
+    @Override
+    public TaskPlan getTaskPlan(InforContext context, TaskPlan taskPlan) throws InforException {
+
+        MP0079_GetTask_001 getTaskPlan = new MP0079_GetTask_001();
+        taskPlan.setTaskRevision(new BigInteger("0"));
```

Comment:
Is Task Revision always 0?

Reply:
Will change the method to set Task Revision to 0 only when not already defined in the Task Plan